src/components/CodeEditor.svelte

```diff
@@ -0,0 +1,123 @@
+<script>
+  import Prism from "prismjs";
+  import "prismjs/components/prism-kotlin";
+  import "prismjs/components/prism-swift";
+  import { afterUpdate } from "svelte"; // eslint-disable-line import/no-extraneous-dependencies
+
+  export let codeTabs = [];
+
+  let codeEditorElement;
+  let currTab = codeTabs[0];
+
+  // language for syntax highlighting is set using the 'language-...' classname so
+  // need to wait for view to be updated before re-highlighting
+  afterUpdate(() => {
+    Prism.highlightElement(codeEditorElement);
+  });
+
+  function handleTabClick(idx) {
+    return function cb() {
+      currTab = codeTabs[idx];
+      codeEditorElement.textContent = currTab.contents;
```

Comment:
Is this needed? the code element should have its content bound to `currTab.contents` in markup already?

Reply:
yes, PrismJS replaces the text content of the code element with new Nodes. Instead of changing this, I will remove `currTab.contents` from the markup to make this clearer.